### ISSUE TITLE
Sample issue for Github Actions

### ISSUE THREAD

**Issue by bytrangle** [bytrangle/most-reacted-comments#1]
This is just a fake issue to test a Github action.

<details id="reacted-comments"><summary><strong>Potentially helpful comments</strong></summary>
<p><a href="https://github.com/bytrangle/most-reacted-comments/issues/1#issuecomment-979059995" rel="nofollow">This project is brand new built with Emotion and N...</p>
<p><a href="https://github.com/bytrangle/most-reacted-comments/issues/1#issuecomment-977602556" rel="nofollow">Can somebody create a preset for this? https://sto...</p>
<p><a href="https://github.com/bytrangle/most-reacted-comments/issues/1#issuecomment-979065068" rel="nofollow">I managed to make it work using TypeScript + Emoti...</p>
</details>

**Comment by bytrangle** [comment]
I'm seemingly bumping into this too. Also have tried all the suggested workarounds in the referenced issues.

My CSS props work just fine outside of Storybook but once I am running in storybook my css props aren't working. Anything that's basically just a string works but any time I am referencing props in the `css / styled` tag, the css property name isn't present.

**Comment by bytrangle** [comment]
I'm seeing this same issue in `@storybook/react@5.1.9`, it makes Storybook's core features basically unusable. I've also attempted all suggested solutions, including `.babelrc` inside `.storybook`, updating `.storybook/webpack.config.js` and others.

Has there been any traction on this issue?

**Comment by bytrangle** [comment]
🙌 `require.resolve('@emotion/babel-preset-css-prop')` + blowing away `node_modules` and rerunning `yarn` fixed it.

Based our now working config kinda on this: emotion-js/emotion#1359 (comment) but with some additional crap for resolving sass stuff (we're in the process of migrating and our storybook stuff stopped working without anyone noticing).

Sorry for any inconvenience and thanks for the awesome library.

**Comment by bytrangle** [comment]
Can somebody create a preset for this? https://storybook.js.org/docs/presets/writing-presets/#docs-content

**Comment by bytrangle** [comment]
I cannot get this to work at all. The only way I can get this to work is if I use the jsx pragma, and then, it breaks storybook's ability to generate or infer prop descriptions for the docs. I'm not certain, but I'm assuming this is because both storybook's doc addon and emotion want to rewrite React.createElement?

**Comment by bytrangle** [comment]
@JakeElder That would be great! 🙇

I created a minimal setup here. This uses CRA + Craco + EmotionJS + Storybook + Cypress. It also has some pinned dependencies to fix issues here and there with CRA + Storybook (this and this).

You can see that if you run the server the Button shown is styled as expected, but it's not within Storybook even if the Babel override is there for pushing @emotion/babel-preset-css-prop. But if you add the pragma in Button.jsx then the style is applied as well in Storybook.

**Comment by bytrangle** [comment]
@se7entyse7en - Reversing the resulting presets array in the resulting Webpack ts/js config seems to make this work as expected. It's a brittle solution but hopefully may help you understand the problem so you can create something more robust. Good luck.

PR: https://github.com/se7entyse7en/js-project-template/pull/1/files

**Comment by bytrangle** [comment]
Yes. This is probably unavoidable in some scenarios until Storybook changes. If you are transpiling with TS (not Babel) try to create custom tsconfig for storybook to disable noUnusedLocals.

**Comment by bytrangle** [comment]
@DominicTobias-b1 Curious — where did you run into missing React in scope? I just checked and none of my files have React in scope and SB compiles and runs just fine. There is one non-obvious place I can think of — shorthand fragment notation <>...</>. If you are using this notation then you would need to import React, but instead you can just import { Fragment } from React; and use <Fragment>...</Fragment>.

**Comment by bytrangle** [comment]
You're right I tried again and worked must have tried in one of the files actually using something off of React. without noticing thanks. Also added 'react/jsx-uses-react': 'off', 'react/react-in-jsx-scope': 'off' for eslint as per React docs.

**Comment by bytrangle** [comment]
Has anyone managed to get Emotion 11 css prop to work in storybook stories files? Haven't managed with old or new pragma with the advice I've read here and elsewhere (and used to work in 10)

**Comment by bytrangle** [comment]
Yes. This is probably unavoidable in some scenarios until Storybook changes. If you are transpiling with TS (not Babel) try to create custom tsconfig for storybook to disable noUnusedLocals.

**Comment by bytrangle** [comment]
@DominicTobias-b1 Curious — where did you run into missing React in scope? I just checked and none of my files have React in scope and SB compiles and runs just fine. There is one non-obvious place I can think of — shorthand fragment notation <>...</>. If you are using this notation then you would need to import React, but instead you can just import { Fragment } from React; and use <Fragment>...</Fragment>.

**Comment by bytrangle** [comment]
It'd be great to have this code encapsulated in a preset!

**Comment by bytrangle** [comment]
With the solutions above I get this in the DOM:

css="You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).
Don't know what I'm missing. I'm using this same Babel configuration for the project build and it works fine.

**Comment by bytrangle** [comment]
We've just encountered this issue. The project uses "typescript": "^4.3.2", "@emotion/react": "^11.4.1", "@storybook/react": "^6.3.8".

tsconfig.json has "jsx": "react-jsx", "jsxImportSource": "@emotion/react", settings.

To fix it, npm i -D @emotion/babel-preset-css-prop and in .storybook/main.js add it to the default presets:

module.exports = {
  stories: […],
  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
  babel: async (options) => {
    options.presets.push('@emotion/babel-preset-css-prop')
    return options
  },
}
So far so good.

**Comment by bytrangle** [comment]
There are 2 solutions:

Using the older @emotion/babel-preset-css-prop.
Modifying Storybook's babel-loader to use a custom configuration.
Both of these versions are super hacky and aren't the correct solution nor will they always be guaranteed to work through Storybook upgrades.

Instead, forget the Webpack config altogether. Storybook has a native way of modifying the Babel config:
https://storybook.js.org/docs/react/configure/babel

The proper option is adding Babel overrides:

// The location of your Babel config.
const babelConfig = require('../babel.config.js');

const storybookConfig = {
  // ...
  babel: async (
    options
  ) => {
    options
      .overrides
      .push({
        ...babelConfig,
        test: '*', // This says "for all files, use this override".
      })

    return options;
  },
  // ...
}

module.exports = storybookConfig

**Comment by bytrangle** [comment]
I fixed it with the settings below.
Hope it helps someone.

pacakge.json

"react": "^17.0.1",
"react-dom": "^17.0.1",
"@storybook/react": "^6.1.21",
"@emotion/react": "^11.1.2",
"typescript": "^4.1.3",
.storybook/main.js

module.exports = {
  reactOptions: {
    fastRefresh: true,
    strictMode: true,
  },
  stories: ['../src/**/*.stories.@(ts|tsx|mdx)'],
  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
  webpackFinal: async (config) => {
    config.module.rules[0].use[0].options.presets = [
      require.resolve('@babel/preset-env'),
      require.resolve('@babel/preset-typescript'),
      [
        require.resolve('@babel/preset-react'),
        {
          runtime: 'automatic',
          importSource: '@emotion/react',
        },
      ],
    ]

    config.module.rules[0].use[0].options.plugins = [
      ...config.module.rules[0].use[0].options.plugins,
      '@emotion/babel-plugin',
    ]

    return config
  },
}

**Comment by bytrangle** [comment]
This project is brand new built with Emotion and Next.js just tonight. I'm surprised I had to add this hack for it to work.

I have Storybook + Emotion playing nicely together without a hack on two separate projects at work, but I'm not sure why it works there and not here.

A few things to note:

I'm on Windows. Dunno if it matters, but maybe it's not able to find the right file.
I named my Babel config file babel.config.js rather than .babelrc.

**Comment by bytrangle** [comment]
I managed to make it work using TypeScript + Emotion 11 and JSX Pragma.

TypeScript: v4.1.4
Emotion: 11.1.5
Storybook: v6.1.17
I'm not using Babel at all because I want to typecheck during build.

I had to change the @babel/preset-react runtime to be "classic" in order to use the JSX Pragma.

**Comment by bytrangle** [comment]
It worked with the following settings.
Just for reference.

**Comment by bytrangle** [comment]
Thanks a lot @illinar 🙏 I had it working but building to old pragma, now I can build with the new one. One thing I had to add to babel.config.js because I'm using TS:

**Comment by bytrangle** [comment]
@jaredh159 @shilman I was able to track at least part of the problem down in the linked storybook-emotion-css repo: https://github.com/emotion-js/emotion/blob/master/packages/babel-plugin-emotion/src/utils/label.js#L66 Seems to be the relevant line -- At least in my testing I was able to get one of the two examples (CssPropButton) rendering correctly by modifying the if to be:

**Comment by bytrangle** [comment]
I also ran into this issue with Babel7+Typescript+React+Emotion+Storybook setup. After hours of digging through docs and experiments I confirmed that .storybook/webpack.config.js:

**Comment by bytrangle** [comment]
@brekk I'm not clear exactly what the fix is you found, and where I would apply it. Could you send a PR to my simple repro repo linked in the OP?

**Comment by bytrangle** [comment]
Also just upgraded to 5.2.0, seeing the same issues.

**Comment by bytrangle** [comment]
in case you missed, I think this ticket is potentially resolvable but I'd ask @ryanlanciaux / @jaredh159 to verify

**Comment by bytrangle** [comment]
Have you tried the babelrc: false option in the loader?

**Comment by bytrangle** [comment]
Thanks. This option seems to do the trick.

**Comment by bytrangle** [comment]
In my project I was able to get Storybook to recognize css-prop (without using the jsx pragma) by adding this config to /.storybook/webpack.config.js.

**Comment by bytrangle** [comment]
if you link the repository or a minimal repro I will take a look for you.

**Comment by bytrangle** [comment]
<details><summary>Potentially helpful comments</summary><p>Epcot is a theme park at Walt Disney World Resort featuring exciting attractions, international pavilions, award-winning fireworks and seasonal special events.</p></details>

**Comment by bytrangle** [comment]
recently upgraded to Emotion 11 with the latest storybook (6.1.xx). Here are the relevant details of my setup. We are using recent Typescript (4.1.x) + Babel (7.12.x), so YMMV depending on your setup.

I switched my components to use new "automatic" React JSX transform. I find this the only reasonable and clean way forward. Note: this requires Typescript 4.1 or higher (see https://emotion.sh/docs/emotion-11#css-prop-types)

**Comment by bytrangle** [comment]
Thanks a lot @illinar 🙏 I had it working but building to old pragma, now I can build with the new one. One thing I had to add to babel.config.js because I'm using TS:

**Comment by bytrangle** [comment]
To make a secret available to an action, you must set the secret as an input or environment variable in the workflow file.

**Comment by jaredh159** [comment]
You can close this as far as I'm concerned -- not because I can verify a solution, but I don't use the css prop at all any more.

**Comment by bytrangle** [comment]
GitHub provides a token that you can use to authenticate on behalf of GitHub Actions.

**Comment by bytrangle** [comment]
Problems parsing JSON

**Comment by bytrangle** [comment]
Python has a built-in package called json, which can be used to work with JSON data.

**Comment by bytrangle** [comment]
UnboundLocalError: local variable 'json' referenced before assignment.

**Comment by bytrangle** [comment]
An action can access the GITHUB_TOKEN through the github.token context even if the workflow does not explicitly pass the GITHUB_TOKEN to the action.

**Comment by bytrangle** [comment]
In Python, we can assign a function to a variable. And using that variable we can call the function as many as times we want. Thereby, increasing code reusability.

**Comment by bytrangle** [comment]
when you perform y=x() you are actually assigning y to the result of calling the function object x and the function has a return value of None. Function calls in python are performed using (). To assign x to y so you can call y just like you would x you assign the function object x to y like y=x and call the function using y().

**Comment by bytrangle** [comment]
Strings in python are surrounded by either single quotation marks, or double quotation marks.

**Comment by bytrangle** [comment]
You can assign a multiline string to a variable by using three quotes.

**Comment by bytrangle** [comment]
Since strings are arrays, we can loop through the characters in a string, with a for loop.

**Comment by bytrangle** [comment]
A variable is only available from inside the region it is created. This is called scope.

**Comment by bytrangle** [comment]
A variable created inside a function belongs to the local scope of that function, and can only be used inside that function.

**Comment by bytrangle** [comment]
[test](https://google.com)

**Comment by bytrangle** [comment]
Virtualenv is another key Python tool. It enables you to create a series of controlled environments where you can install and experiment with Python modules without upsetting any previously installed software.

**Comment by bytrangle** [comment]
When you plan to publish your action to GitHub Marketplace, you'll need ensure that the repository only includes the metadata file, code, and files necessary for the action. Creating a single repository for the action allows you to tag, release, and package the code in a single unit. GitHub also uses the action's metadata on your GitHub Marketplace page.

**Comment by bytrangle** [comment]
Will this work?

**Comment by brekk** [comment]
Hi, not sure why but github is claiming I was mentioned on this issue, any idea why?

![CleanShot 2021-12-16 at 14 37 22](https://user-images.githubusercontent.com/18919/146459348-9ba52920-2262-41e4-b4ec-4caf681d6815.png)
